### PR TITLE
create a more consistant online_id

### PR DESCRIPTION
**Pull request by d-dittmann**
The online_id will generated based on bankcode and accountnumber. The
accountnumber may have leading zeros which shall be ignored to make the
creation reliable. In my case the accountnumber of the accountinfo has no
leading zeros, but the accountnumber of the imported transactions has leading
zeros. In the result the generated online_id was different and the widget
to assign the transactions to a account appears always.

### CODE REVIEW

File: gnucash/import-export/aqb/gnc-ab-utils.c [hunk 1]
```diff
@@ -316,6 +317,28 @@ gnc_AB_VALUE_to_readable_string(const AB_VALUE *value)
...
+static gchar*
+create_online_id(const gchar *bankcode, const gchar *accountnumber)
+{
+    gchar *online_id;
+
+    /* The accountnumber may have leading zeros, depending on where them
+     * accountnumver is came from, e.g. the accountnumber of accountinfo
+     * has no leading zeros while the (local)accountnumber of a transaction
+     * has leading zeros.
+     * So remove all leading '0', to get a consistant online_id.
```
Comment (jsoref) [inline gnucash/import-export/aqb/gnc-ab-utils.c]:
```suggestion
     * So remove all leading '0', to get a consistent online_id.
```

File: gnucash/import-export/aqb/gnc-ab-utils.c [hunk 2]
```diff
@@ -316,6 +317,28 @@ gnc_AB_VALUE_to_readable_string(const AB_VALUE *value)
...
 }
 
+
+static gchar*
+create_online_id(const gchar *bankcode, const gchar *accountnumber)
+{
+    gchar *online_id;
+
+    /* The accountnumber may have leading zeros, depending on where them
+     * accountnumver is came from, e.g. the accountnumber of accountinfo
```
Comment (fellen) [inline gnucash/import-export/aqb/gnc-ab-utils.c]: * accountnumber ...

**Comment by fellen** [comment]
> The change probably solves a bunch of unexpected behaviours for various user, and is unobtrusive enough to just go in. Thanks!

Yes, it is described in https://bugs.gnucash.org/show_bug.cgi?id=797432#c15

**Comment by fellen** [comment]
I fixed the mentioned typos and applied it then.
Thank you!

**Pull request closed without merging**